File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cs
public class Solution {
    public int LengthOfLongestSubstring(string s) {
        HashSet<char> st = new HashSet<char>();
        int res = 0;
        int l = 0;
        for(int r = 0;  r < s.Length; r++)
        {
            while (st.Contains(s[r]))
            {
                st.Remove(s[l]);
                l++;
            }
            st.Add(s[r]);
            res = Math.Max(res, r - l + 1);
        }

        return res;
    }
}

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        // input: array of integers
        // output: length of the longest consective sequence of elements that can be formed.
        int res = 0;
        unordered_set<int> seen(nums.begin(), nums.end());

        for (int num : nums)
        {
            if (!seen.contains(num - 1))
            {
                int length = 1;
                int current = num + 1;
                while (seen.contains(current))
                {
                    length++;
                    current++;
                }

                res = max(res, length);
            }
        }

        return res; 
    }
};
